File: src/lib/overlay.ts
import { invoke } from "@tauri-apps/api/core";

export async function openOverlay(id: string) {
    return invoke("open_overlay", { id });
}

export async function closeOverlay(id: string) {
    return invoke("close_overlay", { id });
}

export async function toggleOverlay(id: string) {
    return invoke("toggle_overlay", { id });
}

export async function focusOverlay(id: string) {
    return invoke("focus_overlay", { id });
}

export async function toggleOverlayDevtools(id: string) {
    return invoke("toggle_overlay_devtools", { id });
}

export async function openCustomCssPath() {
    return invoke("open_custom_css_path");
}

File: src-tauri/src/commands/overlay.rs
use tauri::AppHandle;

use crate::overlays::OverlayStorage;

#[tauri::command]
pub async fn open_overlay(app: AppHandle, id: String) {
    if let Some(mut overlay) = OverlayStorage::get_overlay_by_id(&app, &id.to_string()).await {
        overlay.maybe_open().unwrap();
    }
}

#[tauri::command]
pub async fn close_overlay(app: AppHandle, id: String) {
    if let Some(mut overlay) = OverlayStorage::get_overlay_by_id(&app, &id.to_string()).await {
        overlay.maybe_close().unwrap();
    }
}

#[tauri::command]
pub async fn toggle_overlay(app: AppHandle, id: String) {
    if let Some(mut overlay) = OverlayStorage::get_overlay_by_id(&app, &id.to_string()).await {
        overlay.toggle().unwrap();
    }
}

#[tauri::command]
pub async fn focus_overlay(app: AppHandle, id: String) {
    if let Some(overlay) = OverlayStorage::get_overlay_by_id(&app, &id.to_string()).await {
        overlay.focus().unwrap();
    }
}

#[tauri::command]
pub async fn toggle_overlay_devtools(app: AppHandle, id: String) {
    if let Some(overlay) = OverlayStorage::get_overlay_by_id(&app, &id.to_string()).await {
        overlay.toggle_devtools();
    }
}

#[tauri::command]
pub async fn open_custom_css_path(app: AppHandle) {
    let path = crate::custom_css::path(&app);
    open::that(path).unwrap();
}
